File: samplejava/src/main/java/pyxis/uzuki/live/richutils/samplejava/JSONActivity.java
package pyxis.uzuki.live.richutils.samplejava;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.TextView;

import org.json.JSONObject;

import butterknife.BindView;
import pyxis.uzuki.live.richutilskt.utils.Utils;

/**
 * Created by pyxis on 2017. 6. 12..
 */

public class JSONActivity extends BaseActivity {
    @BindView(R.id.txtResult) TextView txtResult;

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_json);

        String jsonStr = Utils.fileAsString(this.getAssets(), "json", "customer.json");
        JSONObject jsonObject = Utils.createJSONObject(jsonStr);

        if (jsonObject == null) {
            txtResult.setText("Json file isn't given!");
            return;
        }


        String firstName = Utils.getJSONString(jsonObject, "firstName");
        String lastName = Utils.getJSONString(jsonObject, "lastName");
        int age = Utils.getJSONInt(jsonObject, "age");



    }
}

File: sample/src/main/java/pyxis/uzuki/live/richutilssample/JSONActivity.kt
package pyxis.uzuki.live.richutilssample

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_json.*
import org.json.JSONObject
import pyxis.uzuki.live.richutilskt.utils.*

class JSONActivity : AppCompatActivity() {

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_json)

        var jsonStr = this.assets.fileAsString("json", "customer.json")
        var jsonObject: JSONObject? = jsonStr.createJSONObject()

        var firstName = jsonObject?.getJSONString(name = "firstName")
        var lastName = jsonObject?.getJSONString(name = "lastName")
        var age = jsonObject?.getJSONInt(name = "age")

        var address = getJSONObject(jsonObject = jsonObject, name = "address")
        var phoneNumber = getJSONArray(jsonObject, "phoneNumber")

        txtResult.text = "firstName -> $firstName\nlastName -> $lastName\nage -> $age"
    }
}
